uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/LeaveRequestTest.php

```diff
@@ -4366,6 +4367,61 @@ public function testTheTimeForFromAndToDateOfLeaveRequestIsAddedCorrectlyWhenLea
     $this->assertEquals($toDate->format('Y-m-d') ." 23:59:00", $leaveRequest->to_date);
   }
 
+  public function testTheTimeForFromAndToDateOfLeaveRequestStaysUntouchedWhenRequestIsTOIL() {
+    $absenceType = AbsenceTypeFabricator::fabricate([
+      'allow_accruals_request'   => true,
+      'max_leave_accrual'        => 10,
+      // The calculation unit is in days since otherwise a check for
+      // calculation unit in days will take action and this test will be invalid
+      'calculation_unit'         => 1,
+      'allow_accrue_in_the_past' => true
+    ]);
+    $expectedFromDate = '2016-01-08 01:23:45';
+    $expectedToDate = '2016-01-10 12:34:56';
+    $fromDate = new DateTime($expectedFromDate);
+    $toDate = new DateTime($expectedToDate);
+    $contactID = 1;
+    $this->registerCurrentLoggedInContactInSession($contactID);
+
+    $period = AbsencePeriodFabricator::fabricate([
+      'start_date' => $fromDate->format('YmdHis'),
+      'end_date' => $toDate->format('YmdHis'),
+    ]);
+
+    $contract = HRJobContractFabricator::fabricate(
+      ['contact_id' => $contactID],
+      ['period_start_date' => $fromDate->format('Y-m-d')]
+    );
```

Comment:
Thinking about this now, will it be a good idea to have a `instantiateLeaveRequirements` method that creates contract, entitlement and all for tests that only need the requirement with no particular care to date and balance amount and we can put that in a trait and call it anytime we have such tests and then focus on the test proper.
Is that a good idea @davialexandre ?
This is just a comment and not part of the review @igorpavlov

Reply:
@tunbola I think this is an awesome idea, that will reduce the code dup and will increase the clarity of tests suites.